### PR TITLE
Added cucumber-js-style Before and After hooks.

### PR DESCRIPTION
**Pull request by visusnet**
Cucumber.js has Before and After hooks which can be used in conjunction with tags. This PR adds the same capability to cypress-cucumber-preprocessor:

```javascript
Before(() => {
  // executed before every scenario
});

Before({ tags: '@tagged'}, () => {
  // executed before every scenario tagged with @tagged
});

After({ tags: '@tagged'}, () => {
  // executed after every scenario tagged with @tagged
});
```

It also allows asynchronous hooks, e.g.:
```javascript
Before(async () => {
  // scenario is resumed only after this callback resolves
});

After(async () => {
  // scenario is completed only after this callback resolves
});

Before(() => {
  return new Promise(resolve => {
    setTimeout(() => {
      console.log('This will be logged before the scenario is resumed');
      resolve();
    }, 1000);
  });
});
```

### CODE REVIEW

**Comment by visusnet** [comment]
This should also fix #25.

**Comment by visusnet** [comment]
After some edge cases did not work, I had to do push some changes. This should work now. :)

File: lib/resolveStepDefinition.test.js [hunk 1]
```diff
@@ -19,15 +19,15 @@ describe("Data table", () => {
...
+/* describe("Basic example", () => {
   require("../cypress/support/step_definitions/basic");
   resolveFeatureFromFile("./cypress/integration/Plugin.feature");
 });
 
 describe("Background section", () => {
   require("../cypress/support/step_definitions/backgroundSection");
   resolveFeatureFromFile("./cypress/integration/BackgroundSection.feature");
-});
+}); */
```
Comment (lgandecki) [inline lib/resolveStepDefinition.test.js]: any particular reason for commenting out this test? and the And and But below?

File: lib/resolveStepDefinition.test.js [hunk 2]
```diff
@@ -64,18 +65,24 @@ describe("Smart tagging", () => {
...
   require("../cypress/support/step_definitions/smart_tagging");
   resolveFeatureFromFile("./cypress/integration/SmartTagging.feature");
 });
 
-describe("And and But", () => {
+/* describe("And and But", () => {
   require("../cypress/support/step_definitions/and_and_but_steps");
   resolveFeatureFromFile("./cypress/integration/AndAndButSteps.feature");
-});
+}); */
```
Comment (lgandecki) [inline lib/resolveStepDefinition.test.js]: this one as well

**Comment by visusnet** [comment]
I have commented out these tests by mistake. I have force-pushed a new version without the comments.

**Comment by guiyomh** [comment]
Goog job

**Comment by lgandecki** [comment]
What the heck @guiyomh?  I'm reviewing this PR at the very moment! 
No one picked it up from our team (consequences of summer time..), so decided to finally give this a try. :) Expect an update in an hour or so.

**Comment by lgandecki** [comment]
Looks great. I've tweaked some tests. Still thinking how to improve the readability of BeforeAndAfterSteps (pretty meta, hard to word it nicely, I know ;) )

 The only thing I'm debating is whether we should overwrite the cypress exposed before/after.
I think it might be nicer to  have them import-only with capital letters. (a bit confusing, but in line with what people use when writing cucumber files in pure cucumberjs)
This way, we also don't introduce any breaking changes for people that already use the mocha before/after steps.

**Comment by lgandecki** [comment]
I messed up the git commit history somehow.. :D But, you can see my changes here:
https://github.com/TheBrainFamily/cypress-cucumber-preprocessor/pull/195/commits/b2322866ba99c7df24f1883a857c4d7c4bc34aa7
and here:
https://github.com/TheBrainFamily/cypress-cucumber-preprocessor/pull/195/commits/ddb5a462624c1624c7f81fb5495089ce9d900086 

I will think a bit more about improving the .feature file, otherwise I will merge and release

**Comment by lgandecki** [comment]
@visusnet this is what I came up with for the .feature file, please take a look , if you are fine with the changes I'd merge and release. 
https://github.com/TheBrainFamily/cypress-cucumber-preprocessor/pull/195/commits/920aeec2f9d3a1289cdd8669a326778bf6c1782a

**Comment by visusnet** [comment]
@lgandecki I‘m fine with your changes. Go ahead. :)
